solr/solrj/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -798,8 +805,21 @@ public LazyCollectionRef(String collName) {
     @Override
     public synchronized DocCollection get(boolean allowCached) {
       gets.incrementAndGet();
+
+      //we install collection watcher for query aggregator nodes thus fetching from state directly
+      //TODO: below code is for solr nodes; need to look if we can optimize that as well(as we have 30k collections)
+      //that makes zk call every 2 seconds(configurable) to ensure collection state has changed or not.
+      // so frequent zk call vs watches
+      if (ZkStateReader.this.isQueryAggregatorNode) {
```

Comment:
Making sure we don't keep two copy of collection' state.json file. 
1. QA has watch for collections
2. Solr mechanism to fetch collection's state.json. It checks every n-seconds zk, whether collection has updated or not